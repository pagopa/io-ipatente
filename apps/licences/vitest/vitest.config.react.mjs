/// <reference types="vitest" />
/// <reference types="vite/client" />

import react from "@vitejs/plugin-react";
import { loadEnv } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";

export default defineConfig({
  plugins: [react(), tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...configDefaults.coverage.exclude,
        "**/__tests__",
        "src/generated/**",

        // BE
        "src/app/*",
        "src/config/*",
        "src/lib/bff/*",
      ],
      ignoreEmptyLines: true,
      include: ["src/**/*.{tsx,ts}"],
      reporter: ["text", "html"],
    },
    css: true,
    env: loadEnv("test", process.cwd(), ""),
    environment: "jsdom",
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    globals: true,
    include: ["./src/**/*.test.tsx"],
    setupFiles: "./vitest/vitest.setup.react.ts",
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
