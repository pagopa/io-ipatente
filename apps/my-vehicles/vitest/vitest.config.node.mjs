import { configDefaults, defineConfig } from "vitest/config";
import tsconfigPaths from "vite-tsconfig-paths";

export default defineConfig({
  plugins: [tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...(configDefaults.coverage.exclude
          ? configDefaults.coverage.exclude
          : []),
        ".next/*",
        "mocks/*",
        "public/*",
        "*.config.js",
        "src/generated/**",
        "src/config/*",
        "src/types/*",
        "src/main.ts",
        "src/config.ts",
        "src/instrumentation.ts",
        "**/__mocks__/**",
        // FE
        "src/hooks/*",
        "src/layouts/*",
      ],
      extension: configDefaults.coverage.extension
        ?.toString()
        .replace(",.tsx", "")
        .split(","),
      ignoreEmptyLines: true,
      reporter: ["text", "json-summary", "json"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/backend`,
    },
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    include: ["./src/**/__tests__/*.ts"],
    setupFiles: ["./vitest/vitest.setup.node.ts"],
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
