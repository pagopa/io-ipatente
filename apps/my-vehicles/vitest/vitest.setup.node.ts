/**
 * MSW configuration for Vitest (used as setupFile in vitest.config.node.ts)
 */
import { loadEnvConfig } from "@next/env";
import { afterAll, afterEach, beforeAll } from "vitest";

import { mswServerTest } from "../mocks/msw-server";

let mswServer: ReturnType<typeof mswServerTest>;
// Start server before all tests
beforeAll(() => {
  loadEnvConfig(process.cwd());
  // this done to correctly load the environment variables on msw handlers
  mswServer = mswServerTest();
  mswServer.listen({ onUnhandledRequest: "warn" });
});

//  Close server after all tests
afterAll(() => mswServer.close());

// Reset handlers after each test `important for test isolation`
afterEach(() => mswServer.resetHandlers());
