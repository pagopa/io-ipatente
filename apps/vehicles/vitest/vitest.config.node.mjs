import { loadEnv } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";

export default defineConfig({
  plugins: [tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...configDefaults.coverage.exclude,
        "**/__tests__",
        "src/generated/**",
        "src/app/api/auth/mocks/**",
        // FE
        "src/common/*",
        "src/hooks/*",
      ],
      ignoreEmptyLines: true,
      include: ["src/**/*.ts"],
      reporter: ["text", "html"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/backend`,
    },
    env: loadEnv("test", process.cwd(), ""),
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    include: ["./src/**/__tests__/*.ts"],
    setupFiles: ["./vitest/vitest.setup.node.ts"],
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
