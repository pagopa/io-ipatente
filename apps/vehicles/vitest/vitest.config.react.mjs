/// <reference types="vitest" />
/// <reference types="vite/client" />

import react from "@vitejs/plugin-react";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";
import { loadEnv } from "vite";

export default defineConfig({
  plugins: [react(), tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...(configDefaults.coverage.exclude
          ? configDefaults.coverage.exclude
          : []),
        ".next/*",
        "mocks/*",
        "public/*",
        "*.config.js",
        "src/generated/**",
        "src/config/*",
        "src/types/*",
        "src/main.ts",
        "src/config.ts",
        "src/instrumentation.ts",
        "**/__mocks__/**",
        // BE
        "src/app/*",
        "src/lib/be/*",
      ],
      ignoreEmptyLines: true,
      reporter: ["text", "json-summary", "json"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/frontend`,
    },
    css: true,
    env: loadEnv("test", process.cwd(), ""),
    environment: "jsdom",
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    globals: true,
    include: ["./src/**/*.test.tsx"],
    setupFiles: "./vitest/vitest.setup.react.ts",
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
