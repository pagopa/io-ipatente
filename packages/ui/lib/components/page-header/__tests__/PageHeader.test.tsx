import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

import {
  DATA_TEST_ID_PREFIX,
  PageHeader,
  PageHeaderBreadcrumb,
} from "../PageHeader";

const description = "sample-description";
const title = "sample-title";
const breadcrumbs: PageHeaderBreadcrumb[] = [
  { label: "root-page", routePath: "/root" },
  { label: "leaf-page" },
];

describe("Test PageHeader Component", () => {
  it("Should match the snapshot with all props", () => {
    const comp = render(
      <PageHeader
        breadcrumbs={breadcrumbs}
        description={description}
        title={title}
      />,
    );
    expect(comp).toMatchSnapshot();
  });

  it("Should render a main page header", () => {
    render(<PageHeader description={description} title={title} />);

    const pageHeaderElement = screen.queryByTestId(DATA_TEST_ID_PREFIX);
    const pageHeaderTitle = screen.queryByTestId(
      `${DATA_TEST_ID_PREFIX}-title`,
    );
    const pageHeaderDescription = screen.queryByTestId(
      `${DATA_TEST_ID_PREFIX}-description`,
    );
    const pageHeaderBreadcrumbs = screen.queryByTestId(
      `${DATA_TEST_ID_PREFIX}-breadcrumbs`,
    );

    expect(pageHeaderElement).toBeInTheDocument();
    expect(pageHeaderTitle).toBeInTheDocument();
    expect(pageHeaderDescription).toBeInTheDocument();
    expect(pageHeaderBreadcrumbs).not.toBeInTheDocument();

    expect(pageHeaderTitle).toHaveTextContent(title);
    expect(pageHeaderDescription).toHaveTextContent(description);
  });

  it("Should render a details page header", () => {
    const handleBreadcrumbClick = vi.fn();

    render(
      <PageHeader
        breadcrumbs={breadcrumbs}
        onBreadcrumbClick={handleBreadcrumbClick}
        title={title}
      />,
    );

    const pageHeaderElement = screen.queryByTestId(DATA_TEST_ID_PREFIX);
    const pageHeaderTitle = screen.queryByTestId(
      `${DATA_TEST_ID_PREFIX}-title`,
    );
    const pageHeaderDescription = screen.queryByTestId(
      `${DATA_TEST_ID_PREFIX}-description`,
    );
    const pageHeaderBreadcrumbs = screen.queryByTestId(
      `${DATA_TEST_ID_PREFIX}-breadcrumbs`,
    );
    const pageHeaderRootBreadcrumb = screen.queryByTestId(
      `${DATA_TEST_ID_PREFIX}-breadcrumb-root-0`,
    );

    expect(pageHeaderElement).toBeInTheDocument();
    expect(pageHeaderTitle).toBeInTheDocument();
    expect(pageHeaderDescription).not.toBeInTheDocument();
    expect(pageHeaderBreadcrumbs).toBeInTheDocument();
    expect(pageHeaderTitle).toHaveTextContent(title);
    expect(pageHeaderRootBreadcrumb).toBeInTheDocument();

    if (pageHeaderRootBreadcrumb) {
      fireEvent.click(pageHeaderRootBreadcrumb);
      expect(handleBreadcrumbClick).toHaveBeenCalledTimes(1);
      expect(handleBreadcrumbClick).toHaveBeenCalledWith(
        breadcrumbs[0].routePath,
      );
      expect(pageHeaderRootBreadcrumb).toHaveTextContent(breadcrumbs[0].label);
    }
  });

  it("Should not trigger onBreadcrumbClick for a root breadcrumb without routePath", () => {
    const handleBreadcrumbClick = vi.fn();
    const breadcrumbs: PageHeaderBreadcrumb[] = [
      { label: "root-page" },
      { label: "leaf-page" },
    ];

    render(
      <PageHeader
        breadcrumbs={breadcrumbs}
        onBreadcrumbClick={handleBreadcrumbClick}
        title={title}
      />,
    );

    const pageHeaderRootBreadcrumb = screen.queryByTestId(
      `${DATA_TEST_ID_PREFIX}-breadcrumb-root-0`,
    );

    expect(pageHeaderRootBreadcrumb).toBeInTheDocument();

    if (pageHeaderRootBreadcrumb) {
      fireEvent.click(pageHeaderRootBreadcrumb);
      expect(handleBreadcrumbClick).not.toHaveBeenCalled();
      expect(pageHeaderRootBreadcrumb).toHaveTextContent(breadcrumbs[0].label);
    }
  });
});
