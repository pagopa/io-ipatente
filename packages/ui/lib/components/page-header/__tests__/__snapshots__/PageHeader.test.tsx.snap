// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test PageHeader Component > Should match the snapshot with all props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-chstwu-MuiStack-root"
        data-testid="io-ipatente-page-header"
      >
        <nav
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
          data-testid="io-ipatente-page-header-breadcrumbs"
        >
          <ol
            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <button
                class="MuiTypography-root MuiTypography-sidenav MuiLink-root MuiLink-underlineAlways MuiLink-button css-itsvw3-MuiTypography-root-MuiLink-root"
                data-testid="io-ipatente-page-header-breadcrumb-root-0"
              >
                root-page
              </button>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-hehnhb-MuiSvgIcon-root"
                data-testid="ArrowForwardIosIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                />
              </svg>
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                data-testid="io-ipatente-page-header-breadcrumb-leaf"
              >
                leaf-page
              </p>
            </li>
          </ol>
        </nav>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
          data-testid="io-ipatente-page-header-title"
        >
          sample-title
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          data-testid="io-ipatente-page-header-description"
        >
          sample-description
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-chstwu-MuiStack-root"
      data-testid="io-ipatente-page-header"
    >
      <nav
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
        data-testid="io-ipatente-page-header-breadcrumbs"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <button
              class="MuiTypography-root MuiTypography-sidenav MuiLink-root MuiLink-underlineAlways MuiLink-button css-itsvw3-MuiTypography-root-MuiLink-root"
              data-testid="io-ipatente-page-header-breadcrumb-root-0"
            >
              root-page
            </button>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-hehnhb-MuiSvgIcon-root"
              data-testid="ArrowForwardIosIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              data-testid="io-ipatente-page-header-breadcrumb-leaf"
            >
              leaf-page
            </p>
          </li>
        </ol>
      </nav>
      <h3
        class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        data-testid="io-ipatente-page-header-title"
      >
        sample-title
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        data-testid="io-ipatente-page-header-description"
      >
        sample-description
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
